tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,156 @@
+import unittest
+from aiohttp import web
+import aiohttp
+from pathlib import Path
+import asyncio
+import ssl
+
+import logging
+
+import tracemalloc
+tracemalloc.start(40)
+
+
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+logging.basicConfig(level=logging.WARNING)
+
+routes = web.RouteTableDef()
+
+
+class Service:
+
+    async def asyncSetUp(self):
+        log.info(f"{self.__class__.__qualname__}.asyncSetUp")
+
+    async def asyncTearDown(self):
+        log.info(f"{self.__class__.__qualname__}.asyncTearDown")
+
+
+    @staticmethod
+    async def restart(runner):
+        while True:
+            try:
+                for s in list(runner.sites):
+                    log.info(f"Restarting Site {s.name}")
+                    await s.stop()
+                    await s.start()
+            except Exception as e:
+                log.exception(e)
+            finally:
+                await asyncio.sleep(5)
+
+
+    @classmethod
+    def create(cls, app):
+        service = cls()
+        rr = []
+        for r in routes:
+            p = str(Path(r.path).absolute())
+            kwargs = r.kwargs.copy()
+            rr.append(aiohttp.web.RouteDef(method=r.method, path=p, handler=getattr(service, r.handler.__name__),
+                                           kwargs=kwargs))
+        app.add_routes(rr)
+        return service
+
+
+    @routes.get('/{other}', name='other')
+    async def handle_other(self, request):
+        pass
+
+    @routes.get('/', name='root')
+    async def handle_root(self, request):
+        import random
+        off = random.randint(7,17)
+        await asyncio.sleep(off/10)
+        return web.Response(text=f'{off}', status=200)
+
+
+
+class Client:
+    def __init__(self, ctx=None):
+        self.session = aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=ctx))
+
+    async def asyncSetUp(self):
+        log.info(f"{self.__class__.__qualname__}.asyncSetUp")
+
+    async def asyncTearDown(self):
+        log.info(f"{self.__class__.__qualname__}.asyncTearDown")
+        await self.session.close()
+
+    async def aget(self, u):
+        r = await self.session.get(u)
+        return r
+
+    def get(self, u):
+        return self.session.get(u)
+
+def create_setUp_tearDown(s):
+    async def _setUp(app):
+        asyncio.ensure_future(s.asyncSetUp())
+    async def _tearDown(app):
+        asyncio.ensure_future(s.asyncTearDown())
+    return _setUp, _tearDown
+
+
+class TestUnclosedTransport(unittest.IsolatedAsyncioTestCase):
+    def setUp(self):
+        log.info(f"{self.__class__.__qualname__}.setUp")
+        self.loop = asyncio.get_event_loop()
+        self.app = app = web.Application()
+
+        self.service = Service.create(app)
+
+        runner = web.AppRunner(app)
+        self.runner = runner
+        self.server_ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+        self.server_ctx.load_cert_chain('data/service-certificates.pem', 'data/service-key.pem')
```

Comment:
Referencing non-existent files isn't going to work. Also, hardcoding pre-generated files isn't great either. In this test suite, there's good examples of making temporary TLS certificates: we use `trustme` lib for this but you don't need to use it directly because there already a set of pytest fixtures that are ready-to-use — use them instead. Usage example: https://github.com/aio-libs/aiohttp/blob/32833c3/tests/test_client_functional.py#L295-L297.